x-pack/legacy/plugins/ml/server/routes/job_validation.ts

```diff
@@ -0,0 +1,189 @@
+      indexPattern,
+      splitFieldName,
+      query,
+      fieldNames,
+      influencerNames,
+      timeFieldName,
+      earliestMs,
+      latestMs
+    );
+  }
+
+  /**
+   * @apiGroup JobValidation
+   *
+   * @api {post} /api/ml/validate/estimate_bucket_span Estimate bucket span
+   * @apiName EstimateBucketSpan
+   * @apiDescription  Estimates minimum viable bucket span based on the characteristics of a pre-viewed subset of the data
+   */
+  router.post(
+    {
+      path: '/api/ml/validate/estimate_bucket_span',
+      validate: {
+        body: schema.object(estimateBucketSpanSchema),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        let errorResp;
+        const resp = await estimateBucketSpanFactory(
+          context.ml!.mlClient.callAsCurrentUser,
+          context.core.elasticsearch.adminClient.callAsInternalUser,
+          xpackMainPlugin
+        )(request.body)
+          // this catch gets triggered when the estimation code runs without error
+          // but isn't able to come up with a bucket span estimation.
+          // this doesn't return a HTTP error but an object with an error message
+          // which the client is then handling. triggering a HTTP error would be
+          // too severe for this case.
+          .catch((error: any) => {
+            errorResp = {
+              error: true,
+              message: error,
+            };
+          });
+
+        return response.ok({
+          body: errorResp !== undefined ? errorResp : resp,
+        });
+      } catch (e) {
+        // this catch gets triggered when an actual error gets thrown when running
+        // the estimation code, for example when the request payload is malformed
+        throw Boom.badRequest(e);
+      }
+    })
+  );
+
+  /**
+   * @apiGroup JobValidation
+   *
+   * @api {post} /api/ml/validate/calculate_model_memory_limit
+   * @apiName CalculateModelMemoryLimit
+   * @apiDescription Calculates the model memory limit
+   *
+   * @apiSuccess {String} modelMemoryLimit
+   */
+  router.post(
+    {
+      path: '/api/ml/validate/calculate_model_memory_limit',
+      validate: {
+        body: schema.object(modelMemoryLimitSchema),
+      },
+    },
+    licensePreRoutingFactory(xpackMainPlugin, async (context, request, response) => {
+      try {
+        const resp = await calculateModelMemoryLimit(context, request.body);
+
+        return response.ok({
+          body: resp,
+        });
+      } catch (e) {
+        return response.customError(wrapError(e));
+      }
+    })
+  );
+
+  /**
+   * @apiGroup JobValidation
+   *
+   * @api {post} /api/ml/validate/cardinality
+   * @apiName ValidateCardinality
+   * @apiDescription Validates cardinality for the given job configuration
+   */
+  router.post(
+    {
+      path: '/api/ml/validate/cardinality',
+      validate: {
+        body: schema.object(anomalyDetectionJobSchema),
+      },
```

Comment:
(Already present before the PR, but) is there any reason the `anomalyDetectionJobSchema` is not already wrapped with a `schema.object()`? I see that some are and some aren't in `x-pack/legacy/plugins/ml/server/new_platform/anomaly_detectors_schema.ts`